backend/src/helper/index.js

```diff
@@ -1,47 +1,151 @@
-import { propEq } from 'ramda';
-
-export function findInObject(data, searchParameter, exclude) {
-  exclude = exclude || false;
-  return JSON.parse(JSON.stringify(data)).filter(function(parameter) {
-    return Object.keys(searchParameter).every(function(key) {
-      if (!exclude) {
-        return parameter[key] === searchParameter[key];
-      } else {
-        return parameter[key] !== searchParameter[key];
+import {propEq, filter, gte, difference, propSatisfies} from 'ramda';
+
+export async function fetchData(req, opportunities) {
+  const {
+    fullName,
+    gender,
+    email,
+    age,
+    currentCountry,
+    englishLevel,
+    digitalToolsLevel,
+    localLanguageLevel,
+    highestDegreeObtained,
+    educationAndWorkBackground,
+    assessYourJobReadiness,
+    startYourOwnBusiness,
+    gdpr
+  } = await req.body;
+
+  let response = await getCountry(opportunities, currentCountry);
+
+  if (startYourOwnBusiness === 'yes') {
+    return await getEntrepreneurship(response);
+  }
+  response = await excludeEntrepreneurship(response);
+
+  // We have to split the responses to branches here, because at a later time we will have to return to present response
+  const jobBranch = await getJob(response);
+  const eduBranch = await getEducation(response);
+  const trainBranch = await getTraining(response);
+
+  const localJobs = await getLocalJobs(jobBranch, assessYourJobReadiness, educationAndWorkBackground);
+
+  if(localJobs.length) {
+    const sufficientLanguage = await filterByLanguage(localJobs, localLanguageLevel, englishLevel);
+    if(sufficientLanguage.length) {
+      return sufficientLanguage;
+    }
+    return await getLanguageEducation(response);
+  }
+
+  const onlineJobs = await getOnlineJobs(jobBranch, assessYourJobReadiness, educationAndWorkBackground);
+
+  if(onlineJobs.length) {
+    if(digitalToolsLevel >= 7) {
+      if(englishLevel >= 7 ){
+        return onlineJobs;
       }
-    });
-  });
-}
-
-export function fetchData(req) {
-  const data = req.body;
-  return {
-    fullName: data.fullname,
-    age: data.age,
-    gender: data.gender,
-    email: data.email,
-    englishLevel: data.englishlevel,
-    location: data.currentcountry,
-    localLanguageLevel: data.locallanguagelevel,
-    digitalToolsLevel: data.digitaltoolstevel,
-    highestDegreeObtained: data.highestdegreeobtained,
-    assessYourJobReadiness: data.assessyourjobreadiness,
-    gdpr: data.gdpr,
-    filledEntrepreneur: data.hasOwnProperty('startyourownbusiness'),
-    isEntrepreneur:
-      data.hasOwnProperty('startyourownbusiness') &&
-      data.startyourownbusiness === 'yes',
-    filledLocation: data.hasOwnProperty('currentcountry')
-  };
-}
-
-export async function getEntrepreneurship(opportunities) {
+      return await getEnglishEducation(response);
+    }
+    return await getDigitalEducation(response);
+  }
+
+  return [];
+}
+
+export async function filterResponse(opportunities, relation, key, value, exclude = false) {
```

Comment:
@Aeunal  I wonder if we could just use `propEq` instead of sending it as a parameter because it's a globally imported hence you might need to reinvent the wheel I 🤔, but let me know  I am curious to know.

Reply:
Maybe we can but I think we would have to make it as a new method like `filterEqualElements`. Because sometimes I use `filterResponse` to get the elements from the object array that has the same value with the value I gave as a parameter, then I am sending `propEq` as the relation. But if I want to get the elements that have a lower value than I sent, then I am sending `propSatisfies` to check values. I made it this way so the usage of the method would be extendable.

Reply:
@Aeunal, in that case, it makes sense to me to use relation as you did.